openapi.yaml

```diff
@@ -25,254 +37,677 @@ security:
+      parameters:
+        - in: path
+          name: pizzaID
+          required: true
+          type: string
+          description: "ID of pizza to retrieve"
+      responses:
+        200:
+          description: "A single pizza"
+          schema:
+            $ref: "#/definitions/PizzaResult"
+        404:
+          $ref: "#/responses/404"
+        500:
+          $ref: "#/responses/500"
+    patch:
+      description: "Update a pizza recipe"
+      tags:
+        - pizza
+      parameters:
+        - in: path
+          name: pizzaID
           required: true
+          type: string
+          description: "ID of a pizza to retrive"
+        - in: body
+          name: attributes
+          description: "attributes of the pizza to be updated"
           schema:
-            $ref: '#/definitions/PetPostBody'
+            $ref: "#/definitions/PizzaAttributes"
       responses:
-        '201':
-          description: Pet successfully created
+        200:
+          description: "The updated pizza recipe"
           schema:
-            $ref: '#/definitions/PetResult'
-        '400':
-          $ref: '#/responses/400'
-        '409':
-          description: Conflict
+            $ref: "#/definitions/PizzaResult"
+        404:
+          $ref: "#/responses/404"
+        500:
+          $ref: "#/responses/500"
+    delete:
+      description: "Delete a pizza recipe"
+      tags:
+        - pizza
+      parameters:
+        - in: path
+          name: pizzaID
+          required: true
+          type: string
+          description: "ID of a pizza to retrive"
+      responses:
+        200:
+          description: "Returns true is the resource was deleted"
           schema:
-            $ref: '#/definitions/ErrorResult'
-        '500':
-          $ref: '#/responses/500'
-  /pets/{id}:
+            type: boolean
+            example: true
+        404:
+          $ref: "#/responses/404"
+        500:
+          $ref: "#/responses/500"
+
+  /pizzas/{pizzaID}/ingredients:
     get:
-      summary: Info for a specific pet
+      description: "Get the ingredients from the pizza with ID pizzaID"
       tags:
-        - pets
-      description: Get pet by unique ID
-      operationId: getPetById
+        - pizza
+        - ingredient
       parameters:
-        - name: id
-          in: path
-          description: Unique Pet ID, UUID format
+        - in: path
+          name: pizzaID
           required: true
           type: string
+          description: "ID of a pizza to retrive"
       responses:
-        '200':
-          description: Successful response
+        200:
+          description: "A list of ingredients"
           schema:
-            $ref: '#/definitions/PetResult'
-        '404':
-          description: PetResource not found
+            $ref: "#/definitions/IngredientsResult"
+        404:
+          $ref: "#/responses/404"
+        500:
+          $ref: "#/responses/500"
+  /pizzas/{pizzaID}/ingredients/{ingredientID}:
```

Comment:
this (and similar with `/pizzas/{pizzaID/doughs/{doughID}` seemed like the most obvious and simplest way to implement this, but not necessarily the best? reminds me of the conversation about the best way to create/update people's addresses in the persons API

Reply:
Interesting topic. Here're some questions I'd suggest you think about first:

1. Will the `ingredientID` be global unique? In other words, will `/pizzas/1/ingredients/A` and `/pizzas/2/ingredients/A` be the same ingredient or not?
2. How aobut if I want to add a new ingredient to pizza 1? Looks like you didn't have a `POST` method for `/pizzas/{pizzaID}/ingredients`. Correct me if I'm wrong, but I guess the ID you designed should be an auto-increment or a random string which will be generated by the database. If your API do support `POST` for this endpoint, does that mean you'll create repeating records for the same ingredient? In other words, if I want to add `pineapple` to both pizza 1 and pizza 2, I'll need to send two POST requests to `/pizzas/1/ingredients` and `/pizzas/2/ingredients` right? In this case, the auto-generated ingredients won't necessarily be the same. How'd you design your database schema to handle this?
3. If we PATCH the pineapple for pizza 1, will the pineapple in pizza 2 be changed as well? Same thing for the `DELETE`.

Reply:
`ingredientID` is a global unique value, with the idea that you could theoretically do something like finding all pizzas that have pepperoni as an ingredient.  So the workflow to add a new ingredient to a pizza is:

1. If the ingredient doesn't already exist, `POST /ingredients` with the name and type of the new ingredient
2. Add the ingredient to the pizza with `PATCH /pizzas/{pizzaID}/ingredients/{ingredientID}`

If you update the ingredient you'd use the `PATCH ingredients/{ingredientID}` endpoint and not the `PATCH /pizzas/{pizzaID}/ingredients/{ingredientID}`, which is only for adding ingredients to pizzas. `DELETE /pizzas/{pizzaID}/ingredients/{ingredientID}` only removes the ingredient from the pizza. As it stands, you can't delete ingredients from the database.